File: mvpreaderjava/src/main/java/com/example/mvpreaderjava/modle/helper/JUHENewsResponse.java
package com.example.mvpreaderjava.modle.helper;

import android.text.TextUtils;

import java.util.List;

/**
 * Created by sw116 on 2018/4/19.
 */

public class JUHENewsResponse<T> implements ResponseHelper {

    private int error_code;
    private String reason;
    private T result;

    public int getError_code() {
        return error_code;
    }

    public void setError_code(int error_code) {
        this.error_code = error_code;
    }

    public String getReason() {
        return reason;
    }

    public void setReason(String reason) {
        this.reason = reason;
    }

    public T getResult() {
        return result;
    }

    public void setResult(T result) {
        this.result = result;
    }

    @Override
    public boolean success() {
        return error_code == 0;
    }

    static class Result<T> implements ResponseHelper {
        private int stat;
        private List<T> data;

        public int getStat() {
            return stat;
        }

        public void setStat(int stat) {
            this.stat = stat;
        }

        public List<T> getData() {
            return data;
        }

        public void setData(List<T> data) {
            this.data = data;
        }

        @Override
        public boolean success() {
            return stat == 1;
        }
    }

    static class Data implements ResponseHelper {

        private String uniquekey;
        private String title;
        private String date;
        private String category;
        private String author_name;
        private String url;
        private String thumbnail_pic_s;
        private String thumbnail_pic_s02;
        private String thumbnail_pic_s03;

        @Override
        public boolean success() {
            return !TextUtils.isEmpty(uniquekey);
        }

        @Override
        public String toString() {
            return "Data{" +
                    "uniquekey='" + uniquekey + '\'' +
                    ", title='" + title + '\'' +
                    ", date='" + date + '\'' +
                    ", category='" + category + '\'' +
                    ", author_name='" + author_name + '\'' +
                    ", url='" + url + '\'' +
                    ", thumbnail_pic_s='" + thumbnail_pic_s + '\'' +
                    ", thumbnail_pic_s02='" + thumbnail_pic_s02 + '\'' +
                    ", thumbnail_pic_s03='" + thumbnail_pic_s03 + '\'' +
                    '}';
        }
    }
}

File: app/src/main/java/com/savion/mvpreader/model/response/JUHENewsResponse.kt
package com.savion.mvpreader.model.response

import android.text.TextUtils
import com.savion.mvpreader.model.behavior.ResponseHelper

/**
 * Created by savion on 2018/4/10.
 */
class JUHENewsResponse<T> : ResponseHelper {

    var error_code: Int? = null
    var reason: String? = null
    var result: T? = null

    override fun success(): Boolean {
        return error_code == 0
    }

    class JUHENewsResponseResule<T> : ResponseHelper {
        override fun success(): Boolean {
            return stat == 1
        }

        var stat: Int? = null
        var data: T? = null
    }

    class JUHENewsResponseData : ResponseHelper {
        override fun success(): Boolean {
            return !TextUtils.isEmpty(uniquekey)
        }

        var uniquekey: String? = null
        var title: String? = null
        var date: String? = null
        var category: String? = null
        var author_name: String? = null
        var url: String? = null
        var thumbnail_pic_s: String? = null
        var thumbnail_pic_s02: String? = null
        var thumbnail_pic_s03: String? = null
        override fun toString(): String {
            return "JUHENewsResponseData(uniquekey=$uniquekey, title=$title, date=$date, category=$category, author_name=$author_name, url=$url, thumbnail_pic_s=$thumbnail_pic_s, thumbnail_pic_s02=$thumbnail_pic_s02, thumbnail_pic_s03=$thumbnail_pic_s03)"
        }
    }

}